assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -383,9 +383,7 @@ const baseActions = {
 		const { dispatch, select, resolveSelect } = registry;
 
 		const { response: availableAudiences, error: syncError } =
-			yield commonActions.await(
-				dispatch( MODULES_ANALYTICS_4 ).syncAvailableAudiences()
-			);
+			yield fetchSyncAvailableAudiencesStore.actions.fetchSyncAvailableAudiences();
```

Comment:
Thanks @nfmohit, you are right - we should _add_ a call to yield `fetchSyncAvailableCustomDimensionsStore.actions.fetchSyncAvailableCustomDimensions()` and return its error if it returns one, rather than replace the call to `syncAvailableAudiences()`.

Reply:
Thank you @techanvil !

Reply:
@ankitrox, I believe we should revert this change and dispatch the `syncAvailableAudiences` action as before. This change is not necessary.

Additionally, the call to `fetchSyncAvailableCustomDimensions` added in the `syncAvailableAudiences` action should be moved here, right before/after the dispatch to `syncAvailableAudiences`.